bug: the helm application reports an error when it performs an update operation after modifying the configuration file

## Describe the Bug

After installing argo, modify its configuration file, execute `apply` operation, delete process due to namespace delete delay, will report `argocd` namespace already exists, `create` operation will report an error, continue to execute `apply` operation, report `argocd` namespace already exists

The execution process and error log are as follows

```sh
✅ 😁 ./dtm-darwin-arm64 apply -f ../argo.yaml --debug
2022-03-07 22:05:16 ℹ [INFO]  Log level is: debug.
2022-03-07 22:05:16 ℹ [INFO]  Apply started.
2022-03-07 22:05:16 ℹ [DEBUG]  Config file:
---
tools:
- name: argocd
  plugin:
    # name of the plugin
    kind: argocd
    # version of the plugin
    version: 0.2.0
  options:
    # need to create the namespace or not, default: false
    create_namespace: true
    repo:
      # name of the Helm repo
      name: argo
      # url of the Helm repo
      url: https://argoproj.github.io/argo-helm
    # Helm chart information
    chart:
      # name of the chart
      chart_name: argo/argo-cd
      # release name of the chart
      release_name: argocd
      # k8s namespace where ArgoCD will be installed
      namespace: argocd
      # whether to wait for the release to be deployed or not
      wait: true
      # the time to wait for any individual Kubernetes operation (like Jobs for hooks). This defaults to 5m0s
      timeout: 6m
      # whether to perform a CRD upgrade during installation
      upgradeCRDs: true
      # custom configuration (Optional). You can refer to [ArgoCD values.yaml](https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/values.yaml)
      values_yaml: |
        redis:
          image:
            tag: 6.2.6-alpine3.15


2022-03-07 22:05:16 ℹ [INFO]  Using dir <.devstream> to store plugins.
2022-03-07 22:05:16 ℹ [DEBUG]  The global manager m is not initialized.
2022-03-07 22:05:16 ℹ [DEBUG]  Used the Backend: local.
2022-03-07 22:05:16 ℹ [DEBUG]  Got a state from the backend: argocd_argocd -> {argocd {argocd 0.2.0} map[chart:map[chart_name:argo/argo-cd namespace:argocd release_name:argocd timeout:5m upgradeCRDs:true values_yaml:redis:
  image:
    tag: 6.2.6-alpine3.15
 wait:true] create_namespace:true repo:map[name:argo url:https://argoproj.github.io/argo-helm]] map[workflows:deployments:
  - name: argocd-application-controller
    ready: true
  - name: argocd-dex-server
    ready: true
  - name: argocd-redis
    ready: true
  - name: argocd-repo-server
    ready: true
  - name: argocd-server
    ready: true
]}.
2022-03-07 22:05:16 ℹ [DEBUG]  isForce:false
2022-03-07 22:05:16 ℹ [DEBUG]    map[string]interface{}{
  	"chart": map[string]interface{}{
  		"chart_name":   string("argo/argo-cd"),
  		"namespace":    string("argocd"),
  		"release_name": string("argocd"),
- 		"timeout":      string("6m"),
+ 		"timeout":      string("5m"),
  		"upgradeCRDs":  bool(true),
  		"values_yaml":  string("redis:\n  image:\n    tag: 6.2.6-alpine3.15\n"),
  		"wait":         bool(true),
  	},
  	"create_namespace": bool(true),
  	"repo":             map[string]interface{}{"name": string("argo"), "url": string("https://argoproj.github.io/argo-helm")},
  }

2022-03-07 22:05:16 ℹ [DEBUG]  Changes for the plan:
2022-03-07 22:05:16 ℹ [DEBUG]
{
  ActionName: Update,
  Tool: {Name: argocd, Plugin: {Kind: argocd, Version: 0.2.0}}
}
2022-03-07 22:05:16 ℹ [INFO]  Tool < argocd > config drifted from the state, will be updated.
Continue? [y/n]
Enter a value (Default is n): y

2022-03-07 22:05:24 ℹ [INFO]  Start executing the plan.
2022-03-07 22:05:24 ℹ [INFO]  Changes count: 1.
2022-03-07 22:05:24 ℹ [INFO]  -------------------- [  Processing progress: 1/1.  ] --------------------
2022-03-07 22:05:24 ℹ [INFO]  Processing: argocd -> Update ...
2022-03-07 22:05:29 ℹ [INFO]  Uninstalling argocd helm chart.
2022/03/07 22:05:29 uninstall: Deleting argocd
2022/03/07 22:05:32 Starting delete for "argocd-redis" Service
2022/03/07 22:05:32 Starting delete for "argocd-repo-server" Service
2022/03/07 22:05:32 Starting delete for "argocd-server" Service
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" Service
2022/03/07 22:05:32 Starting delete for "argocd-dex-server" Service
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" Deployment
2022/03/07 22:05:32 Starting delete for "argocd-server" Deployment
2022/03/07 22:05:32 Starting delete for "argocd-redis" Deployment
2022/03/07 22:05:32 Starting delete for "argocd-repo-server" Deployment
2022/03/07 22:05:32 Starting delete for "argocd-dex-server" Deployment
2022/03/07 22:05:32 Starting delete for "argocd-dex-server" RoleBinding
2022/03/07 22:05:32 Starting delete for "argocd-server" RoleBinding
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" RoleBinding
2022/03/07 22:05:32 Starting delete for "argocd-dex-server" Role
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" Role
2022/03/07 22:05:32 Starting delete for "argocd-server" Role
2022/03/07 22:05:32 Starting delete for "argocd-server" ClusterRoleBinding
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" ClusterRoleBinding
2022/03/07 22:05:32 Starting delete for "argocd-server" ClusterRole
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" ClusterRole
2022/03/07 22:05:32 Starting delete for "argocd-tls-certs-cm" ConfigMap
2022/03/07 22:05:32 Starting delete for "argocd-gpg-keys-cm" ConfigMap
2022/03/07 22:05:32 Starting delete for "argocd-cm" ConfigMap
2022/03/07 22:05:32 Starting delete for "argocd-ssh-known-hosts-cm" ConfigMap
2022/03/07 22:05:32 Starting delete for "argocd-rbac-cm" ConfigMap
2022/03/07 22:05:32 Starting delete for "argocd-secret" Secret
2022/03/07 22:05:32 Starting delete for "argocd-application-controller" ServiceAccount
2022/03/07 22:05:32 Starting delete for "argocd-dex-server" ServiceAccount
2022/03/07 22:05:32 Starting delete for "argocd-server" ServiceAccount
2022/03/07 22:05:32 purge requested for argocd
2022/03/07 22:05:33 release uninstalled, response: &{0x140007b4070 }
2022-03-07 22:05:33 ℹ [DEBUG]  Prepare to create the namespace: argocd.
2022-03-07 22:05:33 ℹ [DEBUG]  Failed to create the namespace: argocd.
2022-03-07 22:05:33 ℹ [DEBUG]  Start:
argocd_argocd:
  name: argocd
  plugin:
    kind: argocd
    version: 0.2.0
  options:
    chart:
      chart_name: argo/argo-cd
      namespace: argocd
      release_name: argocd
      timeout: 5m
      upgradeCRDs: true
      values_yaml: |
        redis:
          image:
            tag: 6.2.6-alpine3.15
      wait: true
    create_namespace: true
    repo:
      name: argo
      url: https://argoproj.github.io/argo-helm
  resource:
    workflows: |
      deployments:
        - name: argocd-application-controller
          ready: true
        - name: argocd-dex-server
          ready: true
        - name: argocd-redis
          ready: true
        - name: argocd-repo-server
          ready: true
        - name: argocd-server
          ready: true

2022-03-07 22:05:33 ✖ [ERROR]  Plugin argocd Update failed.
2022-03-07 22:05:33 ℹ [DEBUG]  End:
argocd_argocd:
  name: argocd
  plugin:
    kind: argocd
    version: 0.2.0
  options:
    chart:
      chart_name: argo/argo-cd
      namespace: argocd
      release_name: argocd
      timeout: 5m
      upgradeCRDs: true
      values_yaml: |
        redis:
          image:
            tag: 6.2.6-alpine3.15
      wait: true
    create_namespace: true
    repo:
      name: argo
      url: https://argoproj.github.io/argo-helm
  resource:
    workflows: |
      deployments:
        - name: argocd-application-controller
          ready: true
        - name: argocd-dex-server
          ready: true
        - name: argocd-redis
          ready: true
        - name: argocd-repo-server
          ready: true
        - name: argocd-server
          ready: true

2022-03-07 22:05:33 ℹ [INFO]  argocd-Update -> object is being deleted: namespaces "argocd" already exists
2022-03-07 22:05:33 ℹ [INFO]  handle-result -> plugin argocd Update failed
2022-03-07 22:05:33 ✖ [ERROR]  Apply failed => some error(s) occurred during plugins apply process.

```

--- a/internal/pkg/plugin/argocd/update.go
+++ b/internal/pkg/plugin/argocd/update.go
@@ -1,6 +1,8 @@
 package argocd
 
 import (
+	"time"
+
 	"github.com/merico-dev/stream/pkg/util/log"
 )
 
@@ -10,5 +12,7 @@ func Update(options map[string]interface{}) (map[string]interface{}, error) {
 		log.Errorf("Failed to delete the ArgoCD: %s.", err)
 		return nil, err
 	}
+
+	<-time.NewTicker(3 * time.Second).C
 	return Create(options)
 }

--- a/internal/pkg/plugin/jenkins/update.go
+++ b/internal/pkg/plugin/jenkins/update.go
@@ -1,6 +1,8 @@
 package jenkins
 
 import (
+	"time"
+
 	"github.com/merico-dev/stream/pkg/util/log"
 )
 
@@ -11,5 +13,7 @@ func Update(options map[string]interface{}) (map[string]interface{}, error) {
 		log.Errorf("Failed to delete the Jenkins: %s.", err)
 		return nil, err
 	}
+
+	<-time.NewTicker(3 * time.Second).C
 	return Create(options)
 }

--- a/internal/pkg/plugin/kubeprometheus/update.go
+++ b/internal/pkg/plugin/kubeprometheus/update.go
@@ -1,6 +1,8 @@
 package kubeprometheus
 
 import (
+	"time"
+
 	"github.com/merico-dev/stream/pkg/util/log"
 )
 
@@ -11,5 +13,7 @@ func Update(options map[string]interface{}) (map[string]interface{}, error) {
 		log.Errorf("Failed to delete the kube-prometheus: %s", err)
 		return nil, err
 	}
+
+	<-time.NewTicker(3 * time.Second).C
 	return Create(options)
 }

--- a/pkg/util/k8s/static.go
+++ b/pkg/util/k8s/static.go
@@ -42,7 +42,10 @@ func (c *Client) DeleteNamespace(namespace string) error {
 	if namespace == "default" || namespace == "kube-system" {
 		return fmt.Errorf("you can't delete the default or kube-system namespace")
 	}
-	return c.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+
+	gracePeriodSeconds := int64(0)
+	return c.CoreV1().Namespaces().Delete(
+		context.TODO(), namespace, metav1.DeleteOptions{GracePeriodSeconds: &gracePeriodSeconds})
 }
 
 func (c *Client) IsNamespaceExists(namespace string) (bool, error) {

